### PR TITLE
[Mesheryctl] Error coming in version.go (server_spec_handler)  

### LINKED ISSUE #3305: [Chore] Update server spec, session sync handlers to use new Meshkit Errors
Note: This is a child issue of #3275 
#### Current Behavior
Meshery Server (and its other components, like Adapters) use a centralized library, MeshKit, to provide uniform utilities (e.g. raising and logging errors). Currently, Meshery Server uses a now antiquated style of error logging with logrus.

#### Desired Behavior
Ideally, all use of logrus is replaced with use of MeshKit-style error logging.

#### Implementation
Meshery Handlers needs to be:
 - Updated to use meshkit errors. 
 - Switch to handler Log instead of `logrus`

Handlers to update:

- [ ] server spec handler
- [ ]  session sync handler

For reference, you can checkout  the [`grafana handler`](https://github.com/layer5io/meshery/blob/master/handlers/grafana_handlers.go) and [`prometheus handler`](https://github.com/layer5io/meshery/blob/master/handlers/prometheus_handlers.go) or the [docs](https://docs.meshery.io/project/contributing-error).
Define all Error functions for all the handlers to be written in the `handlers/error.go` file.

#### Mockups

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Chadha93 <gauravchadha1676@gmail.com>

**Description**
After making the required changes in the `server_spec_handler` and `error. go`, an error automatically gets raised in the `version. go` **(CheckLatestVersion not declared by package handlers)**

This PR fixes #3305 

**Notes for Reviewers**
This is only for `server_sepc_handler`

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Chadha93** [comment]
@dhruv0000 @sayantan1413 Could you please suggest some changes for `server_spec_handler`.

**Comment by leecalcote** [comment]
Uh-oh... a conflict.

File: handlers/server_spec_handler.go [hunk 1]
```diff
@@ -48,13 +47,14 @@ func (h *Handler) ServerVersionHandler(w http.ResponseWriter, r *http.Request) {
...
-		logrus.Errorf("unable to send data: %v", err)
+		h.log.Error(ErrDataSend(err))
+		http.Error(w, ErrDataSend(err).Error(), http.StatusNotFound)
 	}
 }
 
 // CheckLatestVersion takes in the current server version compares it with the target
 // and returns the result (latest.CheckResponse)
-func CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
+func (h *Handler) CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
```
Comment (sayantan1413) [inline handlers/server_spec_handler.go:57]: Hey @dhruv0000 in this function @Chadha93 is adding the (h *Handler) so that he can use the h.log.Error(..). But this function is generating a problem in the `CheckLatestVersion` which is in the mesheryctl `version.go`. This is because the function `CheckLatestVersion` in mesheryctl does not have this h pointer.  And we cannot import the handler package in the mesheryctl. So me and @Chadha93 was thinking what could be the possible solution for this? You can find this function in mesheryctl version.go in **line number 125**

Comment (dhruv0000) [reply handlers/server_spec_handler.go:57]: The function is using logger to print the info for version it's returning. I think it will make more sense for it to be in `ServerVersionHandler` method and `checkMesheryctlClientVersion` function instead.

**Review by dhruv0000 — CHANGES_REQUESTED** [review]

File: handlers/server_spec_handler.go [hunk 2]
```diff
@@ -67,12 +67,12 @@ func CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
 	}
```
Comment (dhruv0000) [inline handlers/server_spec_handler.go:67]: Update this error too.

**Comment by leecalcote** [comment]
Heads-up: Merge conflict - https://github.com/meshery/meshery/pull/3411/conflicts

**Comment by Chadha93** [comment]
@dhruv0000 I have changed it accordingly but there is again an error coming in `version.go` regarding` checkLatestVersion` not declared. Me and @sayantan1413 also re-checked it but we are not using `CheckLatestVersion` anywhere in `server_spec_handlers.go` now
image

**Comment by dhruv0000** [comment]
> @dhruv0000 I have changed it accordingly but there is again an error coming in `version.go` regarding` checkLatestVersion` not declared. Me and @sayantan1413 also re-checked it but we are not using `CheckLatestVersion` anywhere in `server_spec_handlers.go` now
> image

I guess I explained it wrong. Let me try again. I suggested that, since,
1. `CheckLatestVersion` was used as a generic function so lets not convert it into a handler method,
2. and function is using logger to print info on the version data it's returning, so
you can move this two logs outside the function (and not create it as a handler method) and print them in `ServerVersionHandler` and `checkMesheryctlClientVersion` (FYI they are the methods that called the `CheckLatestVersion` function, I didn't want you to rename `CheckLatestVersion` to `checkMesheryctlClientVersion` ).
```
if res.Outdated {
		logrus.Info("\n  ", serverVersion, " is not the latest Meshery release. Update to v", res.Current, ". Run `mesheryctl system update`")
		h.log.Info(serverVersion, " is not the latest Meshery release. Update to v", res.Current, ". Run `mesheryctl system update`")
	}

	// If user is running the latest release, let them know.
	if res.Latest {
		logrus.Info("\n  ", serverVersion, " is the latest Meshery release.")
		h.log.Info(serverVersion, " is the latest Meshery release.")
	}
```

Poke me if you are still confused.

**Comment by leecalcote** [comment]
Contention - https://github.com/meshery/meshery/pull/3411/conflicts (of course, @dhruv0000 pointed it out faster that I could).
